python/cuda_parallel/cuda/parallel/experimental/iterators/_iterators.py

```diff
@@ -1,35 +1,59 @@
 import ctypes
 import operator
+import uuid
 from functools import lru_cache
 from typing import Dict, Callable
 
 from llvmlite import ir
 from numba.core.extending import intrinsic, overload
 from numba.core.typing.ctypes_utils import to_ctypes
+from numba.cuda.dispatcher import CUDADispatcher
 from numba import cuda, types
 import numba
 import numpy as np
 
+from .._caching import CachableFunction
+
 
 _DEVICE_POINTER_SIZE = 8
 _DEVICE_POINTER_BITWIDTH = _DEVICE_POINTER_SIZE * 8
 
 
+@lru_cache(maxsize=None)
+def _get_abi_suffix(kind: "IteratorKind"):
```

Comment:
I'd move this after `class IteratorKind`: reads better, doesn't need the type hint in quotes.